### DISCUSSION TITLE
Why does the class "prose" shrink elements' widths?

### DISCUSSION THREAD

**Discussion by ooker777** [tailwindlabs/tailwindcss#13455] [Help] [answered]
This code works when there is no class `prose` on `<main>`:
```html
<div class="flex flex-row gap-3 w-full">
  <div class="basis-1/2 bg-yellow-50 p-10">
    123
  </div>  
  <div class="basis-1/2 bg-yellow-50 p-10">
    abc
  </div>  
</div>
```

[![enter image description here][1]][1]

However, the class `prose` from the [Typography](https://github.com/tailwindlabs/tailwindcss-typography "tailwindlabs/tailwindcss-typography: Beautiful typographic defaults for HTML you don't control.") plugin is added:

```diff
-     <main class="flex flex-row gap-3 w-full">
+     <main class="prose flex flex-row gap-3 w-full">
```

Then all element's widths shrink:
[![enter image description here][2]][2]
([Tailwind Play](https://play.tailwindcss.com/uj1vGACRJA?layout=preview)) 

Why is that the case? How to prevent it?

  [1]: https://i.stack.imgur.com/0BrDD.png
  [2]: https://i.stack.imgur.com/vsrm8.png

**Comment by alex-krasikau** [comment, marked as answer]
Hey @ooker777!

By default, the `prose` plugin applies `max-width: 65ch` to a container. This is a good practice for improving the readability of text content.

To remove it, add `max-w-none` to the container.

```diff
- <main class="prose flex flex-row gap-3 w-full">
+ <main class="prose flex w-full flex-row gap-3 max-w-none">
```


You can also achieve the same results by updating the Typography Plugin configuration:
```
module.exports = {
  theme: {
    extend: {
      typography: {
        DEFAULT: {
          css: {
            maxWidth: 'none',
          },
        },
      },
    },
  },
  plugins: [require('@tailwindcss/typography')],
}

```

Here is the complete example: [Play Tailwind - Typography Configuration](https://play.tailwindcss.com/59bkwaLAwJ?file=config).

**Reply by ooker777** [reply]
Thanks. Should this be mentioned in the docs?

**Reply by alex-krasikau** [reply]
This one is already in the docs - [Overriding max-width](https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#overriding-max-width) 😉
